libreoffice-starter-extension-master/source/org/libreoffice/example/dialog/ActionOne.java

```diff
@@ -0,0 +1,370 @@
+			selectedSourceLanguage = sourceLanguages[0];
+			for (int i = 0; i < sourceLanguages.length; i++) {
+				if (sourceLanguages[i].contentEquals(savedSourceLang)) {
+					selectedNr[0] = (short) i;
+					selectedSourceLanguage = savedSourceLang;
+					break;
+				}
+			}
+			sourceLangBoxProps.setPropertyValue("SelectedItems", selectedNr);
+		} catch (IllegalArgumentException | UnknownPropertyException | PropertyVetoException
+				| WrappedTargetException e) {
+			e.printStackTrace();
+		}
+
+		// adjust target language box based on what is set in source box. Similar to upper code.
+		String[] targetLanguages = getTargetLanguageList(selectedSourceLanguage);
+		XControl targetLangBoxControl = m_xControlContainer.getControl("ListBox2");
+		XPropertySet targetLangBoxProps = UnoRuntime.queryInterface(XPropertySet.class, targetLangBoxControl.getModel());
+		try {
+			targetLangBoxProps.setPropertyValue("StringItemList", targetLanguages);
+			short[] selectedNr = new short[] {(short) 0};
+			for (int i = 0; i < targetLanguages.length; i++) {
+				if (targetLanguages[i].contentEquals(savedTargetLang)) {
+					selectedNr[0] = (short) i;
+					break;
+				}
+			}
+			targetLangBoxProps.setPropertyValue("SelectedItems", selectedNr);
+		} catch (IllegalArgumentException | UnknownPropertyException | PropertyVetoException
+				| WrappedTargetException e) {
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Based on given source language retruns non-repeating language list
+	 * with all available target languages for systems that are running.
+	 *
+	 * @param sourceLanguage
+	 * @return	String array with target languages
+	 */
+	private String[] getTargetLanguageList(String sourceLanguage) {
+		SystemSMT[] systems = TildeTranslatorImpl.getSystemList().getSystem();
+
+	    List<String> targetLanguageList = new ArrayList<String>();
+
+	    // update target list: get systems where source language is the selected one
+		for (int i = 0; i < systems.length; i++ ) {
+			String systemsSourceLang = systems[i].getSourceLanguage().getName().getText();
+			String systemsTargetLang = systems[i].getTargetLanguage().getName().getText();
+			// put them in non-repeating array
+			if (systemsSourceLang.equals(sourceLanguage) && (!targetLanguageList.contains(systemsTargetLang))) {
+				// check if machine's status is running
+				for (int k = 0; k < systems[i].getMetadata().length; k++) {
+					String key = systems[i].getMetadata()[k].getKey();
+					if (key.contentEquals("status") && systems[i].getMetadata()[k].getValue().contentEquals("running")) {
+							targetLanguageList.add(systemsTargetLang);
+							break;
+					}
+				}
+			}
+		}
+		targetLanguageList.sort(Comparator.naturalOrder());
+		// change from List<String> to String[]
+		String[] targetLanguageArray = new String[targetLanguageList.size()];
+		for(int i = 0; i < targetLanguageList.size(); i++) {
+			targetLanguageArray[i] = targetLanguageList.get(i);
+		}
+		return targetLanguageArray;
```

Comment:
Javā sarakstus var konvertēt uz masīviem ar `.toArray()` metodes izsaukumu.
Šeit vienkāršāk būs bez for cikla:
`return targetLanguageList.toArray()`